src/advanced_inv.cpp

```diff
@@ -1338,7 +1338,7 @@ bool advanced_inventory::move_all_items( bool nested_call )
                 if( !query_yn( _( "Really drop all your favorite items?" ) ) ) {
                     return false;
                 }
-                dropped = dropped_favorite;
+                dropped = std::move( dropped_favorite );
```

Comment:
```suggestion
                dropped.swap( dropped_favorite );
```
Move semantics are confusing and fragile, this has the same results performance-wise and is easier to reason about.